v2/api/insights/v1api20201001/activity_log_alert_types_gen.go

```diff
@@ -0,0 +1,2440 @@
+}
+
+// OriginalGVK returns a GroupValueKind for the original API version used to create the resource
+func (alert *ActivityLogAlert) OriginalGVK() *schema.GroupVersionKind {
+	return &schema.GroupVersionKind{
+		Group:   GroupVersion.Group,
+		Version: alert.Spec.OriginalVersion(),
+		Kind:    "ActivityLogAlert",
+	}
+}
+
+// +kubebuilder:object:root=true
+// Generator information:
+// - Generated from: /monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json
+// - ARM URI: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/activityLogAlerts/{activityLogAlertName}
+type ActivityLogAlertList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ActivityLogAlert `json:"items"`
+}
+
+type ActivityLogAlert_Spec struct {
+	// +kubebuilder:validation:Required
+	// Actions: The actions that will activate when the condition is met.
+	Actions *ActionList `json:"actions,omitempty"`
+
+	// AzureName: The name of the resource in Azure. This is often the same as the name of the resource in Kubernetes but it
+	// doesn't have to be.
+	AzureName string `json:"azureName,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// Condition: The condition that will cause this alert to activate.
+	Condition *AlertRuleAllOfCondition `json:"condition,omitempty"`
+
+	// Description: A description of this Activity Log Alert rule.
+	Description *string `json:"description,omitempty"`
+
+	// Enabled: Indicates whether this Activity Log Alert rule is enabled. If an Activity Log Alert rule is not enabled, then
+	// none of its actions will be activated.
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// Location: The location of the resource. Azure Activity Log Alert rules are supported on Global, West Europe and North
+	// Europe regions.
+	Location *string `json:"location,omitempty"`
+
+	// OperatorSpec: The specification for configuring operator behavior. This field is interpreted by the operator and not
+	// passed directly to Azure
+	OperatorSpec *ActivityLogAlertOperatorSpec `json:"operatorSpec,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// Owner: The owner of the resource. The owner controls where the resource goes when it is deployed. The owner also
+	// controls the resources lifecycle. When the owner is deleted the resource will also be deleted. Owner is expected to be a
+	// reference to a resources.azure.com/ResourceGroup resource
+	Owner *genruntime.KnownResourceReference `group:"resources.azure.com" json:"owner,omitempty" kind:"ResourceGroup"`
+
+	// +kubebuilder:validation:Required
+	// ScopesReferences: A list of resource IDs that will be used as prefixes. The alert will only apply to Activity Log events
+	// with resource IDs that fall under one of these prefixes. This list must include at least one item.
+	ScopesReferences []genruntime.ResourceReference `armReference:"Scopes" json:"scopesReferences,omitempty"`
```

Comment:
question that will probably be answered as I finish the review: What ARM type goes here?

Reply:
Oh it looks like StorageAccounts here works, for example?

Reply:
Yes. I'm not sure what other kinds of resources are permitted.